File: 1950-sign-of-the-product-of-an-array/sign-of-the-product-of-an-array.dart
class Solution {
  int arraySign(List<int> nums) {    
    var negativeCount = 0;
    for (final num in nums) {
      if (num == 0) {
        return 0;
      } else if (num.sign == -1) {
        negativeCount++;
      }
    }
    return negativeCount.isEven ? 1 : -1;
  }
}

File: 1950-sign-of-the-product-of-an-array/sign-of-the-product-of-an-array.go
func arraySign(nums []int) int {
    negativeCount := 0

    for _, num := range nums {
        if num == 0 {
            return 0
        } else if num < 0 {
            negativeCount++
        }
    }

    if negativeCount%2 == 0 {
        return 1
    } else {
        return -1
    }
}
